### PR TITLE
Fix incorrect errno

### PR DESCRIPTION
**Pull request by lherman-cs**

### CODE REVIEW

**Comment by at-wat** [comment]
I think clearing `errno` is a caller's responsibility. Manpage says:

http://man7.org/linux/man-pages/man3/errno.3.html
> The value of errno is never set to zero by any system call or library function.

**Comment by lherman-cs** [comment]
Hmm, that's interesting. This becomes really strange when we have to set `errno` from Go instead. What do you think?

**Comment by at-wat** [comment]
I'm guessing that something is actually failing (or warning) inside the cpp functions.
Clearing errno at the beginning of the cpp functions which is called from Go would be safer.

**Comment by lherman-cs** [comment]
I see. That makes sense actually. I'll update the PR

File: pkg/codec/openh264/bridge.cpp [hunk 1]
```diff
@@ -51,8 +54,10 @@ Encoder *enc_new(const EncoderOptions opts) {
     errno = rv;
     return NULL;
   }
+  printf("3: %d\n", errno);
```
Comment (lherman-cs) [inline pkg/codec/openh264/bridge.cpp]: So, somehow the errno got set to 60 (Operation timeout) here. But, when I tried to ignore this errno, it worked just fine.

Comment (at-wat) [reply pkg/codec/openh264/bridge.cpp]: I found that errno is not goroutine safe on Darwin.
```go
package main
  
import (
  "fmt"
  "net"
  "time"
)

// #include <unistd.h>
// #include <errno.h>
// int wait() { usleep(2000000); }
import "C"

func main() {
  for i := 0; i < 10; i++ {
    go func() {
      for {
        _, _ = net.Dial("tcp", "localhost:12345")
        time.Sleep(time.Millisecond)
      }
    }()
  }
  _, err := C.wait()
  fmt.Printf("err: %v\n", err)
}
```
This code outputs `nil` on Linux, but on Darwin (tested on Travis-CI's OSX environment) outputs `operation timed out`.

Comment (at-wat) [reply pkg/codec/openh264/bridge.cpp]: So, it seems unsafe to use errno on Darwin.
How about https://github.com/pion/mediadevices/pull/88?

Comment (lherman-cs) [reply pkg/codec/openh264/bridge.cpp]: I always thought that the cgo error handling is pretty neat, using the second value for error by implying from errno. But, nevermind, it's not safe.... Oh, well, I like the approach, I don't know if there's a cleaner way of doing it atm.

**Comment by lherman-cs** [comment]
Closing this in favor of https://github.com/pion/mediadevices/pull/88

**Pull request closed without merging**